File: Jobtrekpro/postman/con.js
// Utility helpers
function pick(arr) { return arr[Math.floor(Math.random() * arr.length)]; }

function randomDOB() {
    const today = new Date();
    const years = Math.floor(Math.random() * 45) + 21; // 21 to 65 years old
    today.setFullYear(today.getFullYear() - years);
    today.setDate(today.getDate() - Math.floor(Math.random() * 365));
    return today.toISOString().split("T")[0];
}

// Data pools
const FIRST = [
    'John', 'Michael', 'David', 'James', 'Robert', 'William',
    'Sarah', 'Jennifer', 'Lisa', 'Mary', 'Patricia', 'Linda',
    'Ahmed', 'Ali', 'Hassan', 'Fatima', 'Aisha', 'Zainab',
    'Chris', 'Daniel', 'Matthew', 'Emma', 'Olivia', 'Sophia',
    'Illana', 'Carlos', 'Marcus', 'Elena', 'Diego', 'Isabella', 'Nina'
];

const LAST = [
    'Smith', 'Johnson', 'Williams', 'Brown', 'Jones', 'Garcia',
    'Miller', 'Davis', 'Rodriguez', 'Martinez', 'Hernandez',
    'Khan', 'Ahmed', 'Ali', 'Hassan', 'Malik', 'Shah',
    'Wilson', 'Moore', 'Taylor', 'Anderson', 'Thomas', 'Jackson',
    'Dickson', 'Cooper', 'Bennett', 'Foster', 'Coleman', 'Brooks'
];

const ADDRESSES = [
    "123 Main Street", "456 Oak Avenue", "789 Elm Boulevard",
    "321 Pine Road", "654 Maple Drive", "987 Cedar Lane",
    "147 Birch Street", "258 Willow Way", "369 Aspen Court",
    "741 Spruce Avenue"
];

const TAG_NAMES = [
    "urgent", "vip", "priority", "certified", "active",
    "verified", "expert", "reliable", "preferred", "new-hire"
];

const TAG_COLORS = [
    '#FF6B6B', '#4ECDC4', '#45B7D1', '#FFA07A', '#98D8C8',
    '#F7DC6F', '#BB8FCE', '#85C1E2', '#F8B195', '#C06C84',
    '#6C5B7B', '#355C7D', '#F67280', '#C8E6C9', '#FFCCBC',
    '#0055ff', '#ff5500', '#00ff55', '#5500ff', '#ff0055'
];

const SPECIALIZATIONS = [
    "electrical", "plumbing", "hvac", "carpentry",
    "painting", "roofing", "cleaning", "landscaping"
];

const SKILLS = ["beginner", "intermediate", "advanced", "expert"];
const VEHICLES = ["car", "truck", "van", "suv"];

// Generate random first and last name
let first = pick(FIRST);
let last = pick(LAST);

// Email based on first + last (no numbers)
let email = `${first.toLowerCase()}.${last.toLowerCase()}@yopmail.com`;

// Random fields
let hasVehicle = Math.random() < 0.3;

// Set Postman variables
pm.variables.set("first", first);
pm.variables.set("last", last);
pm.variables.set("email", email);

pm.variables.set("phone", `+1 (${Math.floor(100 + Math.random() * 900)}) ${Math.floor(1000000 + Math.random() * 9000000)}`);
pm.variables.set("dateOfBirth", randomDOB());
pm.variables.set("address", pick(ADDRESSES));

pm.variables.set("serviceRadius", pick([10, 15, 25, 30, 50]));
pm.variables.set("hourlyRate", Math.floor(Math.random() * 80) + 20); // 20-100
pm.variables.set("yearsExperience", Math.floor(Math.random() * 40) + 1); // 1-40

pm.variables.set("skillLevel", pick(SKILLS));
pm.variables.set("specialization", pick(SPECIALIZATIONS));

pm.variables.set("hasVehicle", hasVehicle);
pm.variables.set("vehicleType", hasVehicle ? pick(VEHICLES) : "");

pm.variables.set("ownsTools", Math.random() < 0.5);

pm.variables.set("tagName", pick(TAG_NAMES));
pm.variables.set("tagColor", pick(TAG_COLORS));

/* Example body json


{
    "firstName": "{{first}}",
    "lastName": "{{last}}",
    "email": "{{email}}",
    "phone": "{{phone}}",
    "dateOfBirth": "{{dateOfBirth}}",
    "homeBaseAddress": "{{address}}",
    "serviceRadius": {{serviceRadius}},
    "hourlyRate": {{hourlyRate}},
    "yearsExperience": {{yearsExperience}},
    "skillLevel": "{{skillLevel}}",
    "specializations": [
        "{{specialization}}"
    ],
    "hasVehicle": {{hasVehicle}},
    "vehicleType": "{{vehicleType}}",
    "ownsTools": {{ownsTools}},
    "serviceAreas": [],
    "status": "pending",
    "tags": [
        {
            "name": "{{tagName}}",
            "color": "{{tagColor}}"
        }
    ]
} */

File: Jobtrekpro/con.py
import requests
import random
import time
from datetime import datetime, timedelta

# === CONFIGURATION ===
API_URL = "https://api.jobtrekpro.com/api/contractors"
ADMIN_TOKEN = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOiJlOWIwNDk0Ny05MTdmLTQ1YzQtYWQ3NS1jNjQxZjZhZDAwMmQiLCJlbWFpbCI6Im1hcmtAeW9wbWFpbC5jb20iLCJyb2xlIjoib3duZXIiLCJpYXQiOjE3NjQzMjI3NjQsImV4cCI6MTc2NDQwOTE2NH0.2ZH7BGpB2ym6tZfe7cyxJIXTb6EFyejRUirzCx6B3tc"
BASE_EMAIL = "con@yopmail.com"

# Request headers
headers = {
    "Authorization": f"Bearer {ADMIN_TOKEN}",
    "Content-Type": "application/json"
}

# Random data pools
first_names = [
    'John', 'Michael', 'David', 'James', 'Robert', 'William',
    'Sarah', 'Jennifer', 'Lisa', 'Mary', 'Patricia', 'Linda',
    'Ahmed', 'Ali', 'Hassan', 'Fatima', 'Aisha', 'Zainab',
    'Chris', 'Daniel', 'Matthew', 'Emma', 'Olivia', 'Sophia',
    'Illana', 'Carlos', 'Marcus', 'Elena', 'Diego', 'Isabella'
]

last_names = [
    'Smith', 'Johnson', 'Williams', 'Brown', 'Jones', 'Garcia',
    'Miller', 'Davis', 'Rodriguez', 'Martinez', 'Hernandez',
    'Khan', 'Ahmed', 'Ali', 'Hassan', 'Malik', 'Shah',
    'Wilson', 'Moore', 'Taylor', 'Anderson', 'Thomas', 'Jackson',
    'Dickson', 'Cooper', 'Bennett', 'Foster', 'Coleman'
]

# Tag options
TAG_NAMES = [
    "urgent", "vip", "priority", "certified", "active",
    "verified", "expert", "reliable", "preferred", "new-hire"
]

TAG_COLORS = [
    '#FF6B6B', '#4ECDC4', '#45B7D1', '#FFA07A', '#98D8C8',
    '#F7DC6F', '#BB8FCE', '#85C1E2', '#F8B195', '#C06C84',
    '#6C5B7B', '#355C7D', '#F67280', '#C8E6C9', '#FFCCBC',
    '#0055ff', '#ff5500', '#00ff55', '#5500ff', '#ff0055'
]

# Specializations
SPECIALIZATIONS = [
    "electrical", "plumbing", "hvac", "carpentry", "painting",
    "roofing", "cleaning", "landscaping"
]

# Skill levels - THIS WILL BE RANDOMLY SELECTED EACH TIME
SKILL_LEVELS = ["beginner", "intermediate", "advanced", "expert"]

# Vehicle types
# Removed None - will handle separately
VEHICLE_TYPES = ["car", "truck", "van", "suv"]

# Home base addresses
ADDRESSES = [
    "123 Main Street",
    "456 Oak Avenue",
    "789 Elm Boulevard",
    "321 Pine Road",
    "654 Maple Drive",
    "987 Cedar Lane",
    "147 Birch Street",
    "258 Willow Way",
    "369 Aspen Court",
    "741 Spruce Avenue"
]


def generate_random_dob():
    """Generate random date of birth (age between 21-65)"""
    today = datetime.now()
    years_ago = random.randint(21, 65)
    dob = today - timedelta(days=years_ago * 365 + random.randint(0, 364))
    return dob.strftime("%Y-%m-%d")


def generate_contractor(i):
    """Generates a contractor payload with random data"""
    first = random.choice(first_names)
    last = random.choice(last_names)

    # Generate unique email using Yopmail alias
    base_name = BASE_EMAIL.split("@")[0]
    domain = BASE_EMAIL.split("@")[1]
    email = f"{base_name}+{i}@{domain}"

    # SELECT ONLY ONE TAG (randomly each time)
    selected_tag_name = random.choice(TAG_NAMES)
    tags = [{"name": selected_tag_name, "color": random.choice(TAG_COLORS)}]

    # SELECT ONLY ONE SPECIALIZATION (randomly each time)
    specializations = [random.choice(SPECIALIZATIONS)]

    # RANDOMLY SELECT SKILL LEVEL (this happens fresh for each contractor)
    skill_level = random.choice(SKILL_LEVELS)

    # Random boolean values
    has_vehicle = random.choice([True, False])
    owns_tools = random.choice([True, False])

    # If has vehicle, always select a vehicle type (not None)
    vehicle_type = random.choice(VEHICLE_TYPES) if has_vehicle else None

    # Create the contractor payload
    payload = {
        "firstName": first,
        "lastName": last,
        "email": email,
        "phone": f"+1 ({random.randint(100, 999)}) {random.randint(1000000, 9999999)}",
        "dateOfBirth": generate_random_dob(),
        "homeBaseAddress": random.choice(ADDRESSES),
        "serviceRadius": random.choice([10, 15, 25, 30, 50]),
        "hourlyRate": random.randint(20, 100),
        "yearsExperience": random.randint(1, 40),
        "skillLevel": skill_level,  # Using the randomly selected skill level
        "specializations": specializations,
        "hasVehicle": has_vehicle,
        # Will be a valid type if hasVehicle=True, None otherwise
        "vehicleType": vehicle_type,
        "ownsTools": owns_tools,
        "tags": tags,
        "serviceAreas": [],
        "licenseNumber": None,
        "licenseState": None,
        "insuranceProvider": None,
        "emergencyContactName": None,
        "emergencyContactPhone": None
    }

    return payload


def create_contractors(n=10):
    """Creates n contractors and sends the data to the API"""
    success, fail = 0, 0

    print(f"\n🚀 Creating {n} contractors...")
    print("=" * 80)

    for i in range(1, n + 1):
        payload = generate_contractor(i)
        try:
            # Send the POST request to create a contractor
            response = requests.post(
                API_URL, json=payload, headers=headers, timeout=10)

            if response.status_code in [200, 201]:
                success += 1
                tag_name = payload['tags'][0]['name']
                spec_name = payload['specializations'][0]
                print(
                    f"[{i}] ✅ {payload['firstName']} {payload['lastName']} | {payload['email']}")
                print(
                    f"     📋 {payload['skillLevel']} | {payload['yearsExperience']}yrs | ${payload['hourlyRate']}/hr")
                print(f"     🏷️  Tag: {tag_name} | Spec: {spec_name}")
            else:
                fail += 1
                print(f"[{i}] ❌ Failed ({response.status_code}): {response.text}")
        except Exception as e:
            fail += 1
            print(f"[{i}] ⚠️ Exception: {e}")

        # Small delay to avoid rate limits
        time.sleep(0.1)

    print("=" * 80)
    print(f"\n✨ Completed! ✅ Success: {success} | ❌ Failed: {fail}")
    print(
        f"📬 All emails accessible at: https://yopmail.com/{BASE_EMAIL.split('@')[0]}")


if __name__ == "__main__":
    # Create 20 contractors for testing
    create_contractors(20)

    # For larger batches:
    # create_contractors(50)
    # create_contractors(100)
